pyrokinetics/cli/convert.py

```diff
@@ -0,0 +1,162 @@
+from argparse import ArgumentParser, Namespace
+from pathlib import Path
+from textwrap import dedent
+
+from pyrokinetics import Pyro
+
+description = "Convert a gyrokinetics input file to a different code."
+
+
+def add_arguments(parser: ArgumentParser) -> None:
+    parser.add_argument(
+        "target",
+        type=str,
+        help="The target gyrokinetics code, e.g. 'GS2', 'CGYRO'.",
+    )
+
+    parser.add_argument(
+        "input_file",
+        type=Path,
+        help="The gyrokinetics config file you wish to convert.",
+    )
+
+    parser.add_argument(
+        "--input_type",
+        type=str,
+        help="The code type of the input file. If not provided, this will be inferred.",
+    )
+
+    parser.add_argument(
+        "--geometry",
+        "-g",
+        type=str,
+        help=dedent(
+            """\
+            The type of flux surface geometry to convert to. Options include 'Miller' or
+            'Fourier'.
```

Comment:
Technically there are a lot more now though I'm not sure if this is best place to put it but we have the following that are fully working
Miller (all)
MillerTurnbull (gene)
MXH (cgyro, tglf)

The fourier ones haven't been implemented in the `gk_code`, we can do the fit but haven't mapped it to the codes yet.